.github/workflows/docker.yml

```diff
@@ -79,8 +84,21 @@ jobs:
       - name: List built images
         run: docker images
 
+      - name: Install graphviz dependencies
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -y graphviz libgraphviz-dev
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip --quiet install --no-cache-dir .[local]
+          python -m pip list
+
       - name: Check basic run works
-        run: >-
-          docker run --rm
-          recast/recastatlas:sha-${GITHUB_SHA::8}
-          -c "recast --help; recast run testing/busyboxtest --backend local"
+        env:
+          RECAST_DOCKER_IMAGE: recast/recastatlas:sha-${GITHUB_SHA::8}
+        run: |
+          docker info
+          recast --help
+          recast run testing/busyboxtest --backend docker
```

Comment:
@lukasheinrich It seems that the `docker` backend is having issues

```pytb
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/recastatlas/subcommands/run.py", line 56, in run
    run_sync(name, spec, backend=backend)
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/recastatlas/backends/__init__.py", line 76, in run_sync
    BACKENDS[backend].run_workflow(name, spec)
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/recastatlas/backends/docker.py", line 111, in run_workflow
    subprocess.check_call(command)
  File "/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '-i', '-v', '/home/runner/work/recast-atlas/recast-atlas:/home/runner/work/recast-atlas/recast-atlas', '-w', '/home/runner/work/recast-atlas/recast-atlas', '-v', '/var/run/docker.sock:/var/run/docker.sock', '-e', 'PACKTIVITY_CVMFS_LOCATION=/cvmfs', '-e', 'PACKTIVITY_CVMFS_PROPAGATION=rprivate', '-e', 'PACKTIVITY_AUTH_LOCATION=None', '-e', 'YADAGE_SCHEMA_LOAD_TOKEN=None', 'recast/recastatlas:sha-${GITHUB_SHA::8}', 'sh', '-c', 'mkdir -p ~/.docker\necho \'{}\' > ~/.docker/config.json\ncat << \'EOF\' | yadage-run -f -\n{"dataarg": "recast-04371ab4", "dataopts": {}, "initdata": {}, "workflow": "rootflow.yml", "toplevel": "github:yadage/yadage-workflows:testing/busybox-flow", "visualize": true, "backend": "multiproc:auto"}\nEOF\n']' returned non-zero exit status 125.
```

Though I'm not fully sure why as I'm able to get things to run locally

```bash
#!/bin/bash

export RECAST_DOCKER_IMAGE="recast/recastatlas:debug"
recast --help
recast run testing/busyboxtest --backend docker
```
```console
$ bash docker_backend.sh 
Usage: recast [OPTIONS] COMMAND [ARGS]...

Options:
  -l, --loglevel TEXT
  --help               Show this message and exit.

Commands:
  auth       Authentication Commands (to gain access to internal data)
  backends   The RECAST computational backends.
  catalogue  The RECAST Analysis Catalogue
  ci         Helper Commands for CI systems
  retrieve   Retrieve RECAST Results from asynchronous submissions
  run        Run a RECAST Workflow synchronously
  software   Build Container Images for RECAST
  status     Get the Status of a asynchronous submission
  submit     Submit a RECAST Workflow asynchronously
  tests      Run a test
2021-09-20 15:23:15,560 | packtivity.asyncback |   INFO | configured pool size to 12
2021-09-20 15:23:15,943 |      yadage.creators |   INFO | no initialization data
2021-09-20 15:23:15,943 |    adage.pollingexec |   INFO | preparing adage coroutine.
2021-09-20 15:23:15,943 |                adage |   INFO | starting state loop.
2021-09-20 15:23:16,027 |     yadage.wflowview |   INFO | added </hello:0|defined|unknown>
2021-09-20 15:23:16,797 |     yadage.wflowview |   INFO | added </world:0|defined|unknown>
2021-09-20 15:23:17,698 |    adage.pollingexec |   INFO | submitting nodes [</hello:0|defined|known>]
2021-09-20 15:23:17,699 |                adage |   INFO | unsubmittable: 0 | submitted: 0 | successful: 0 | failed: 0 | total: 2 | open rules: 0 | applied rules: 2
2021-09-20 15:23:17,700 |      pack.hello.step |   INFO | starting file logging for topic: step
2021-09-20 15:23:19,603 |           adage.node |   INFO | node ready </hello:0|success|known>
2021-09-20 15:23:19,603 |    adage.pollingexec |   INFO | submitting nodes [</world:0|defined|known>]
2021-09-20 15:23:19,605 |      pack.world.step |   INFO | starting file logging for topic: step
2021-09-20 15:23:21,548 |           adage.node |   INFO | node ready </world:0|success|known>
2021-09-20 15:23:21,569 | adage.controllerutil |   INFO | no nodes can be run anymore and no rules are applicable
2021-09-20 15:23:21,569 | adage.controllerutil |   INFO | no nodes can be run anymore and no rules are applicable
2021-09-20 15:23:21,569 |                adage |   INFO | unsubmittable: 0 | submitted: 0 | successful: 2 | failed: 0 | total: 2 | open rules: 0 | applied rules: 2
2021-09-20 15:23:27,591 |                adage |   INFO | adage state loop done.
2021-09-20 15:23:27,591 |                adage |   INFO | execution valid. (in terms of execution order)
2021-09-20 15:23:27,591 |                adage |   INFO | workflow completed successfully.
2021-09-20 15:23:27,591 |  yadage.steering_api |   INFO | done. dumping workflow to disk.
2021-09-20 15:23:27,593 |  yadage.steering_api |   INFO | visualizing workflow.
/usr/lib/python3.8/subprocess.py:848: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used
  self.stdout = io.open(c2pread, 'rb', bufsize)
/usr/lib/python3.8/subprocess.py:848: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used
  self.stdout = io.open(c2pread, 'rb', bufsize)
/usr/lib/python3.8/subprocess.py:848: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used
  self.stdout = io.open(c2pread, 'rb', bufsize)
/usr/lib/python3.8/subprocess.py:848: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used
  self.stdout = io.open(c2pread, 'rb', bufsize)
2021-09-20 10:23:28,254 | recastatlas.subcomma |   INFO | RECAST run finished.

RECAST result testing/busyboxtest recast-ac3ea92e:
--------------
[]

```

Reply:
this seemss to be too deeeply nested docker (note it mounts docker.sock which might not be avaliable on GHA)

Reply:
https://github.com/maxheld83/ghactions/issues/307

Reply:
Okay I'll try to follow up more on this later.

Reply:
yeah for now I think it's fine if we test `--local`

Reply:
Yeah, you mean running `recast run testing/busyboxtest --backend local` inside of the image that was just built? Testing this locally just failed on me, so maybe something that also requires more debugging in general.

Reply:
Moving this to Issue #50